### PR TITLE
Fix #3519 handle invalid card details with Authorize.net

### LINKED ISSUE #3519: Problems checking out with Authorize.net when you enter an invalid card
### Expected behavior
If you enter an invalid card you should get a meaningful error message and a chance to retry

### Actual behavior
If you enter a card that's the right characters but is invalid you get a `Server Error (500)` message. If you enter a card that is not the right number of characters you get the correct validation message but it sticks on "Submitting" and doesn't allow you to re-enter unless you refresh

#### Scenario One
1. Login as admin and enable shipping and Authorize.net
1. Try to checkout but for credit card number put "4040404040404040"
1. Observe you get `Server Error`

#### Scenario Two
1. Login as admin and enable shipping and Authorize.net
1. Try to checkout but for credit card number put "12345"
1. Observe that the "Submit Payment" button just says "Submitting" and never allows you to reinput your card number

### Versions
```
Node: 8.9.1
NPM: 5.5.1
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.25.0
Reaction: 1.6.4
Reaction branch: master
Docker: 17.12.0-ce
```

### PR DESCRIPTION
**Pull request by Akarshit**
Fixes #3519

### To test
1. Login as admin and enable shipping and Authorize.net
1. Try to checkout but for credit card number put "12345"
1. Observe you get normal field validation error.
1. Try credit card number "4040404040404040" and click "Complete your order"
1. Observe that you get `Error: Invalid card details [400]`

### CODE REVIEW

File: imports/plugins/included/payments-authnet/server/methods/authnet.js [hunk 1]
```diff
@@ -98,9 +102,12 @@ Meteor.methods({
...
       throw new Meteor.Error("invalid-transaction-type", "Invalid Transaction Type");
     }
-
-    const result =  Promise.await(authResult);
-    return result;
+    try {
+      const result =  Promise.await(authResult);
+      return result;
+    } catch (error) {
+      throw new Meteor.Error(400, error.message);
```
Comment (zenweasel) [inline imports/plugins/included/payments-authnet/server/methods/authnet.js]: So passing in this number here is deprecated. Please check the style guide but basically it should be in the format
`Meteor.Error("short-description", "Detailed description")`

File: imports/plugins/included/payments-authnet/server/methods/authnet.js [hunk 2]
```diff
@@ -62,12 +62,16 @@ const ValidCVV = Match.Where(function (x) {
...
-    });
+    try {
+      check(cardInfo, {
+        cardNumber: ValidCardNumber,
+        expirationYear: ValidExpireYear,
+        expirationMonth: ValidExpireMonth,
+        cvv2: ValidCVV
+      });
+    } catch (error) {
+      throw new Meteor.Error(400, "Invalid card details");
```
Comment (zenweasel) [inline imports/plugins/included/payments-authnet/server/methods/authnet.js]: So passing in this number here is deprecated. Please check the style guide but basically it should be in the format
`Meteor.Error("short-description", "Detailed description")`

File: imports/plugins/included/payments-authnet/server/methods/authnet.js [hunk 3]
```diff
@@ -91,9 +95,12 @@ Meteor.methods({
...
       throw new Meteor.Error("invalid-transaction-type", "Invalid Transaction Type");
     }
-
-    const result = Promise.await(authResult);
-    return result;
+    try {
+      const result =  Promise.await(authResult);
+      return result;
+    } catch (error) {
+      throw new Meteor.Error("auth-failed", error.message);
```
Comment (aaronjudd) [inline imports/plugins/included/payments-authnet/server/methods/authnet.js]: @zenweasel @machikoyasuda we could really use a list of approved error messages, and standardization.. do we have these documented?

Comment (machikoyasuda) [reply imports/plugins/included/payments-authnet/server/methods/authnet.js]: Added an issue: https://github.com/reactioncommerce/reaction-docs/issues/513

**Pull request merged**